deploy/cdk/lib/maintain-metadata/maintain-metadata-stack.ts

```diff
@@ -160,14 +160,100 @@ export class MaintainMetadataStack extends Stack {
+
+        #set($subjects = $ctx.stash.itemRecord.subjects)
+        $!{ctx.stash.put("subjectsBefore", $subjects)}
+
+        #set($keys = [])
+
+    		#foreach($subject in $subjects)
+          #set($map = {})
+          #set($uri = $util.str.toUpper($util.defaultIfNullOrBlank($subject.uri, "")))
+          #if ( $uri != "" )
+            $util.qr($map.put("PK", $util.dynamodb.toString("SUBJECTTERM")))
+            $util.qr($map.put("SK", $util.dynamodb.toString("URI#$uri")))
+            $util.qr($keys.add($map))
+          #end
+		    #end
+
+        ## Query all records based on the primary key
+
+        {
+            "version" : "2017-02-28",
+            "operation" : "BatchGetItem",
+            "tables": {
+              "${websiteMetadataTable.tableName}": {
+                "keys": $util.toJson($keys),
+                "consistentRead": true
+              },
+            },
+        }`),
+      responseMappingTemplate: MappingTemplate.fromString(`
+        ## Raise a GraphQL field error in case of a datasource invocation error
+        #if($ctx.error)
+            $util.error($ctx.error.message, $ctx.error.type)
+        #end
+        ## Pass back the result from DynamoDB. **
+        ### Add extra processing here to try to generate a single record of output
+        
+        #set($subjectsAfter = [])
+        ## First, add subjects from database query
+        #foreach($item in $context.result.data.${websiteMetadataTable.tableName})
+          #set($map = {})
+          #foreach( $entry in $util.map.copyAndRemoveAllKeys($item, ["PK","SK","TYPE","GSI1PK","GSI1SK","GSI2PK","GSI2SK","dateAddedToDynamo","dateModifiedInDynamo"]).entrySet() )
+            ## $!{results.put("$entry.key", "$entry.value")}
+            #set($map[$entry.key] = $entry.value)
+          #end
+          $util.qr($subjectsAfter.add($map))
+        #end
+
+        ## Next, add in subjects that were not found in the query
+        #foreach($subject in $ctx.stash.subjectsBefore)
+          #if ( $util.defaultIfNullOrBlank($subject.uri, "") == "")
+            $util.qr($subjectsAfter.add($subject))
+          #else
+            #set($subjectInAfterList = 0)
+            #set($uriToFind = $util.str.toUpper($util.defaultIfNullOrBlank($subject.uri, "")))
+            #foreach($subjectAfter in $ctx.stash.subjectsAfter)
+              #if ( $uriToFind == $util.str.toUpper($util.defaultIfNullOrBlank($subjectAfter.uri, "")) )
+                #set($subjectInAfterList = 1)
+              #end
+              #if ( $subjectInAfterList == 0 )
+                $util.qr($subjectsAfter.add($subject))
+              #end
+            #end
+          #end
+        #end
+
+        ## Finally, replace existing subjects in record with new replacements
+        #set($itemRecord = $ctx.stash.itemRecord)
+        ## $!{itemRecord.put("subjects", $subjectsAfter)}
+        #set($itemRecord["subjects"] = $subjectsAfter)
+        ## $!{ctx.stash.put("subjectsAfter", $subjectsAfter)}
```

Comment:
Do we need this large commented(?) block?

Reply:
The individual # are actually part of the velocity programming language code.  The ## are comments.

Reply:
gotcha